src/main/java/com/hazelcast/aws/impl/DescribeInstances.java

```diff
@@ -76,30 +73,48 @@ public DescribeInstances(AwsConfig awsConfig, String endpoint) throws IOExceptio
         addFilters();
     }
 
+    //Just for testing purposes
     DescribeInstances(AwsConfig awsConfig) {
-        if (awsConfig == null) {
-            throw new IllegalArgumentException("AwsConfig is required!");
-        }
         this.awsConfig = awsConfig;
     }
 
-    void checkKeysFromIamRoles(Environment env) throws IOException {
-        if (awsConfig.getAccessKey() != null && awsConfig.getIamRole() == null) {

```

Comment:
This check has been moved to `execute` method. `fillKeysFromIamRoles` is not called when `accessKey` is set and IAM role is not set. see [here](https://github.com/hazelcast/hazelcast-aws/pull/31/files#diff-81b776ac33a4d0d722dde3ec4d9377e3R247)